### ISSUE TITLE
Await expression can not be a class heritage

### ISSUE THREAD

**Issue by JLHwung** [KFlash/seafox#12]
```js
async function f() {   class x extends await y { }   }
```

Per spec: https://tc39.es/ecma262/#prod-ClassHeritage `ClassHeritage` should be a LeftHandSideExpression. If I am reading correctly, AwaitExpression is _not_ a LeftHandSideExpression.

**Comment by KFlash** [comment]
Good catch. You are right! I will try to fix it soon as I find time for it. Working on a new language atm. Want to join?

**Comment by JLHwung** [comment]
@KFlash Which language is your next target? I might take a look but most of my time are still spent on JS.

**Comment by KFlash** [comment]
It's js. I'm trying to create a language like TS that will compile down to Js. But with improved performance ofc. And plan to add in linting and minify options
